### PR TITLE
Add EC2 module Configurations

### PR DESCRIPTION
**Pull request by Debashbora**
Added EC2 module:
image
Screenshot from 2024-05-14 12-10-12
Screenshot from 2024-05-14 12-10-04
Screenshot from 2024-05-14 12-09-41

### CODE REVIEW

File: modules/ec2/main.tf [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  }
+}
+resource "aws_security_group" "instance_sg" {
+  name        = "instance_sg"
+  description = "Security group for EC2 instance"
+
+  vpc_id = var.vpc_id
+
+  ingress {
+    from_port   = 22
```
Comment (vijay-stephen) [inline modules/ec2/main.tf]: This has to be a variable

File: modules/ec2/main.tf [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+provider "aws" {
+  region = var.region
+}
+data "aws_ami" "amazon-ami" {
+  most_recent = true
+  owners      = ["amazon"]
+
+  filter {
+    name   = "name"
```
Comment (vijay-stephen) [inline modules/ec2/main.tf]: This has to be a variable

File: modules/ec2/main.tf [hunk 3]
```diff
@@ -0,0 +1,74 @@
+terraform {
+  required_version = "~> 1.4"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+  }
+  #   backend "s3" {}
```
Comment (vijay-stephen) [inline modules/ec2/main.tf]: Remove this.

**Review by mayank0202 — APPROVED** [review]
Currently vpc , subnets and ami are hardcoded and thats fine for now , but we will use data lookups whenever we  call this `modules/ec2` in `terraform` directory

Looks LGTM

**Pull request merged**